### ISSUE TITLE
Few questions about triplet loss 

### ISSUE THREAD

**Issue by TCHeish** [JDAI-CV/fast-reid#242]
Hi,

According to your hard_example_mining function in triplet_loss.py, I saw you comment out few lines:

    # ap_weight = F.softmax(pos_dist, dim=1)
    # dist_ap = torch.sum(ap_weight * pos_dist, dim=1)
    ...
    # an_weight = F.softmax(-neg_dist, dim=1)
    # dist_an = torch.sum(an_weight * neg_dist, dim=1)

This is based on "Features for Multi-Target Multi-Camera Tracking and Re-Identification", they weighted each sample
instead of pick the hardest one, right?
And I wonder that since you comment out the codes, is that because it didn't work ?

Thanks!

**Comment by L1aoXingyu** [comment]
It's moving to `weighted_example_mining`

https://github.com/JDAI-CV/fast-reid/blob/1d6e970260c8638688ac0a4976c1f1fbb7b3b6ee/fastreid/modeling/losses/triplet_loss.py#L122
